mininet/node.py

```diff
@@ -845,18 +845,31 @@ class XIAHost( Host ):
     def __init__( self, name, **kwargs ):
         pathCheck( 'xip', moduleName='XIA' )
         Node.__init__( self, name, **kwargs )
-        self.xids = {}  # mapping of configurable inputs to XID types
 
-    def config( self, hid=None, xdp=False, **params):
-        """hid: private key file names for Host Identifiers, eg. hid=[ 'pk1','pk2' ]
-           xdp: set True to load the xdp module.
-           params: parameters for Node.config()."""
+    def config( self, ad=None, hid=None, lpm=None,
+                u4id=None, zfid=None, xdp=False, **params ):
+        """hid: private key file names for Host Identifiers.
+           zfid: zFilter identifier.
+           xdp: set True to load the xdp module
+           params: parameters for Node.config()"""
         r = Node.config( self, **params )
         if xdp is True:
             moduleDeps( add='xia_ppal_xdp' )
+        self.setParam( r, 'setAD', ad=ad )
         self.setParam( r, 'setHID', hid=hid )
+        self.setParam( r, 'setLPM' , lpm=lpm )
+        self.setParam( r, 'setU4ID', u4id=u4id )
+        self.setParam( r, 'setZFID', zfid=zfid )
         return r
 
+    def setAD( self, *ad ):
+        """Assign a host to Atonomous Domain.
+           Similar to BGP's Atonomous System.
+           ad:list of ad identifiers. """
+        moduleDeps( add='xia_ppal_ad' )
+        for a in ad:
+            self.cmd( 'xip ad addlocal', a )
```

Comment:
It's a good idea to add small error logs than giving the user the entire usage or real bulky error messages. Moreover, our terse logs would help the user identify where the configurations went wrong. So that the user can configure the XIDs after the network eventually starts. But, raising an exception ain't a good idea. An exception won't let the network start.

All we want the user to know is where exactly the configurations could not be completed and if possible, the corresponding IDs will be printed on screen as well.